src/main.tsx

```diff
@@ -9,33 +9,56 @@ import {
   Welcome,
   Login,
   RegistrationConfirmation,
-  MedicationReminderForm,
   Settings,
 } from "./routes";
-import { client } from "./store";
+
+import ProtectedRoute from "./components/protected-route/protected-route";
+
+const client = new ApolloClient({
+  uri: "http://localhost:8080/api/graphql",
+  cache: new InMemoryCache(),
+});
 
 ReactDOM.createRoot(document.getElementById("root")!).render(
   <React.StrictMode>
     <ApolloProvider client={client}>
       <BrowserRouter>
         <Routes>
           <Route path="/" element={<div>Home Route</div>} />
+          <Route path="/welcome" element={<Welcome />} />
           <Route
             path="/registration-confirm"
-            element={<RegistrationConfirmation />}
+            element={
+              <ProtectedRoute>
+                <RegistrationConfirmation />
```

Comment:
There are 2 ways to do registration. 
1. The user is logged in and needs to confirm the email
2. The user is not logged in, needs to confirm the email and then needs to login to gain access to the site.

I think the latter is more secure, so this route shouldn't be protected. 
Speak to Josh to make sure.